crates/red_knot_python_semantic/resources/mdtest/type_of/any.md

```diff
@@ -0,0 +1,43 @@
+# type[Any]
+
+## Simple
+
+```py
+def f(x: type[Any]):
+    reveal_type(x)  # revealed: type[Any]
+    reveal_type(x.__repr__)  # revealed: Any
```

Comment:
I think this could actually be `Literal[__repr__] & Any` -- it could be a subclass of `type` with a narrower definition of `__repr__`, but it can't be wider than `object.__repr__`. (Though as discussed below, the `Literal[__repr__]` part of this isn't really correct, we just don't handle bound methods yet.) But I think all of this can be TODO pending more handling of `Type::SubclassOf` attributes.

```suggestion
    # TODO could be `<object.__repr__ type> & Any`
    reveal_type(x.__repr__)  # revealed: Any
```